pyptex/__init__.py

```diff
@@ -532,31 +551,32 @@ def input(self,filename,argv=False):
         that is private to `a.tex`, much like each Python module has a private global
         scope. In a similar fashion, `b.tex` has its own private global scope.
         The global `pyp` objects in `a.tex` and `b.tex` are also different instances
-        of the `pyptex` class. This is similar to the notion of "compilation units" in 
+        of the `pyptex` class. This is similar to the notion of "compilation units" in
         the C programming language.
 
         If one wishes to pass some parameters from `a.tex` to `b.tex`, one may use
-        the notation `pyp.input('b.tex',argv)`, which will initialize the global
+        the notation `pyp.input('b.tex', argv)`, which will initialize the global
         `pyp` object of `b.tex` so that it contains the field `pyp.argv=argv`.
 
         If one absolutely needs to export variables from `b.tex` back to `a.tex`, one
-        should directly use the `pyptex` constructor, e.g. `pyp_b = pyptex('b.tex',argv)`;
+        should directly use the `pyptex` constructor, e.g. `pyp_b = pyptex('b.tex', argv)`;
         one can then retrieve values from the `b.tex` scope, e.g. with `pyp_b.fragments[0]`.
         """
-        ret=pyptex(filename,argv or self.argv,False)
-        return r"\input{"+ret.pyptexfilename+"}"
-    def open(self,filename,*argv,**kwargs):
-        """If pyp = pyptex('a.tex') then pyp.open(filename,...) is a wrapper for
-        the builtin function open(filename,...) that further adds filename to
+        ret = pyptex(filename, argv or self.argv, False)
+        return fr'\input{{{ret.pyptexfilename}}}'
+
+    def open(self, filename, *argv, **kwargs):
+        """If pyp = pyptex('a.tex') then pyp.open(filename, ...) is a wrapper for
+        the builtin function open(filename, ...) that further adds filename to
         the list of dependencies via pyp.dep(filename).
         """
         self.dep(filename)
-        return open(filename,*argv,**kwargs)
+        return open(filename, *argv, **kwargs)
 
-def pyptexmain(argv=None):
-    """
-    This function parses an input file a.tex to produce a.pyptex and a.pdf, by
-    doing pyp = pyptex('a.tex',...) object. The filename a.tex must be in argv[1];
+
+def pyptexmain(argv: List[str] = None):
```

Comment:
This is a quick question here. I agree *this* argv must be a List, but I wonder if it has to be a List[str]. I didn't unit test something like argv=["a.tex",17]

By the way, although I am being a bit picky about types of argv, I should stay I am strongly in favor of using types as much as possible. I'm not being a luddite, ideally I'd type everything and use the Python typechecker even, whose name escapes me, to find even more bugs in my code. So thanks for getting started on the type annotations.

Reply:
Oops, being gung-ho with the type annotations again. I'll change this to just one `list`.